### DISCUSSION TITLE
Use pre-alpha (snapshot) builds of compose-jb?

### DISCUSSION THREAD

**Discussion by twyatt** [JuulLabs/krayon#154] [Polls]
Krayon is currently stuck on an older Kotlin version (1.6.10 at the time of this writing) because of [compose-jb].

This may become troublesome for consumers, as libraries such as [kotlinx.datetime] have specific Kotlin version compatibility. For example, starting with Kotlin 1.7.0, [kotlinx.datetime] 0.3.3+ is required. If there are breaking API changes in upcoming [kotlinx.datetime] releases, then Krayon could cause runtime issues for people if they are using a newer (incompatible) [kotlinx.datetime] to that which Krayon is compiled against.

In order to use Kotlin 1.7.0, Krayon would need to switch to using pre-alpha (snapshot) builds of [compose-jb], such as mentioned in https://github.com/JetBrains/compose-jb/issues/2108#issuecomment-1172229775.

[kotlinx.datetime]: https://github.com/Kotlin/kotlinx-datetime
[compose-jb]: https://github.com/JetBrains/compose-jb

**Comment by cedrickcooke** [comment]
I'm fine building on a pre-alpha, the old libraries have become painful. We'll just have to be careful not releasing (or only releasing a `-prerelease` build) until things progress.

**Comment by twyatt** [comment]
This has been addressed via #156.
